testing/kfctl/kfctl_go_test.py

```diff
@@ -65,6 +65,18 @@ def test_build_kfctl_go(app_path, project, use_basic_auth, use_istio, config_pat
   run_with_retries(["make", "build-kfctl"], cwd=build_dir)
   kfctl_path = os.path.join(build_dir, "bin", "kfctl")
 
+  # Push built kfctl binary to GCS bucket
+  import google.cloud.storage as storage
+  gcs_client = storage.Client()
+  bucket = gcs_client.lookup_bucket("kfctl-nightly")
```

Comment:
Should we just use a GCS bucket and then do different subpaths?
We could also just copy kfctl to the bucket that is used by prow.
Then the binaries would show up in the artifacts tab of spyglass e.g.
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/kubeflow_kubeflow/3856/kubeflow-presubmit/1165454527155933184/

And people could get to them just by navigating to the appropriate pre/post/periodic  run.